### ISSUE TITLE
DemoBench hangs/freezes when used in conjunction with full-screen PowerPoint presentations and projector on Mac

### ISSUE THREAD

**Issue by gendal** [corda/corda#1049]
This is a really annoying problem that hits me pretty much EVERY time I try to use DemoBench for its primary purpose:  demonstrating Corda in a presentation context.  Unfortunately, I can't reproduce it in the office :(

Here's what happens:

* I plug an HDMI connector into my MacBook at a customer venue and check that the PowerPoint slides I'm going to give project OK.  

* They do!   And the resolution is pretty good... wow... somebody in the world has a set-up that supports more than VGA.  Things are looking good!

* I then fire up DemoBench and start some nodes so that I don't have to wait for them to start up when I get to the demo portion of my talk.  I usually also fire up some Node Explorers and issue some cash and position them in the right position on the screen and check the network map viewer is the right size, etc, etc, .... all is good.  This is going to be a KILLER presentation!!  :) 

* I leave all that running, start my PowerPoint slideshow and leave the title screen up while I wait for the room to fill.  This is a presentation about Corda from R3's  See Tee Oh...  wow..!! room is standing room only... people are cheering....  I'm wearing my best black turtle-neck...   it's like the launch of a new iPhone!!

* I spend a few minutes giving some introductory remarks and work through a few slides.

* I then reach the slide that promises the demo...  the moment the WHOLE WORLD has been waiting for...  you can FEEL the anticipation...!!!   The billions of people watching the live stream are frozen...  they don't dare look away in case they miss even a SECOND of what is about to be revealed...!

* I hit "esc".... I click on the DemoBench window to bring it to the foreground...  only one more second until the best demonstration in the HISTORY. OF. THE. WORLD!!!

* Oh :(

* The DemoBench UI is completely frozen and unresponsive.  

* Disaster.

* Thwarted again.

* The Node Explorers (and the nodes themselves) still work thankfully... so I can do _part_ of the demo... 

* But the moment has passed.  

So...   my question for this issue is...   if I can't reproduce it in the office, what diagnostics can/should I collect in the field?

**Comment by chrisr3** [comment]
If the UI has become unresponsive then it sounds as if either there's been an exception in the FX application thread, or possibly the FX application thread is "stuck" waiting for something. In the former case, I would hope that _something_ has been captured in `$HOME/demobench/demobench.log`. In the latter case, the only think I can think of would be

    $ jstack <PID of demobench>

to locate the guilty operation on the stack trace.

**Comment by gendal** [comment]
Woo!!! I reproduced it!!

[demobench.exception.txt](https://github.com/corda/corda/files/1148509/demobench.exception.txt)
[jstack.txt](https://github.com/corda/corda/files/1148510/jstack.txt)

**Comment by gendal** [comment]
I just needed to leave PowerPoint fullscreen/in the foreground for long enough I guess!

I guess the lesson is: I need to be less long-winded and do my demo earlier in my sessions :)

**Comment by chrisr3** [comment]
And that exception is being thrown from within Swing; there is no `net.corda.*` code on that stack trace. Which suggests that disabling App Nap may indeed be the only solution.

**Comment by chrisr3** [comment]
I've raised [issue #8](https://github.com/JetBrains/jdk8u/issues/8) for the JetBrains JDK. Rick has suggested that this may be triggered by "App Nap".

**Comment by gendal** [comment]
somebody else reported something tantalisingly similar...   https://stackoverflow.com/questions/40378247/an-exception-in-mouseinfo-getpointerinfo-getlocation    feels like it could turn into one of these...  https://xkcd.com/979/

**Comment by chrisr3** [comment]
Yes indeed, `JLightweightFrame` looks like this, and the exception is from line 473;
```java
472  private void updateClientCursor() {
473      Point p = MouseInfo.getPointerInfo().getLocation();
```

**Comment by mikehearn** [comment]
I'm not quite sure what the connection to AppNap is but if that's the cause we can easily work around it (or used to be able to) with an entry in the Info.plist file.

**Comment by chrisr3** [comment]
Fixed by #1069.

**Comment by gendal** [comment]
AppNap was a red herring.
